core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt

```diff
@@ -129,4 +149,12 @@ abstract class Gene(var name: String) {
      */
     abstract fun containsSameValueAs(other: Gene): Boolean
 
+    /**
+     * indicates if the gene reaches its optimal value.
+     */
+    open fun reachOptimal() = false
```

Comment:
I need more explanation here...
How can a gene be "optimal"?
the optimality depends on testing target, and a gene could involved in the reaching of hundreds of targets (if not even more)